src/marvin/components/ai_function.py

```diff
@@ -1,105 +1,85 @@
 import asyncio
 import functools
-import inspect
-import re
-from typing import Callable, TypeVar
+from typing import Callable, TypeVar, Union
 
 from pydantic import BaseModel
 
 from marvin.engine.executors import OpenAIExecutor
-from marvin.prompts import library as prompt_library
+from marvin.engine.language_models import ChatLLM
+from marvin.functions import Function
+from marvin.prompts import Prompt, render_prompts
+from marvin.prompts.library import System, User
 from marvin.tools.format_response import FormatResponse
 from marvin.utilities.async_utils import run_sync
 from marvin.utilities.types import safe_issubclass
 
 T = TypeVar("T")
 A = TypeVar("A")
 
-prompts = [
-    prompt_library.System(content="""
+
+class FunctionSystem(System):
+    content = """\
+        {% if prefix %}
+        {{prefix}}
+        
+        {% endif %}
+        {% if source_code %}
         Your job is to generate likely outputs for a Python function with the
         following signature and docstring:
-    
-        {{ function_def }}        
-        
-        The user will provide function inputs (if any) and you must respond with
-        the most likely result. 
-        
-        {% if function_description %}
-        The following function description was also provided:
 
-        {{ function_description }}
+        {{ source_code }}    
+        
         {% endif %}
+        The user will provide function inputs (if any) and you must respond with
+        the most likely result.\
+        {% if description %}
         
-        ## Response Format
         
-        Your response must match the function's return signature. To validate your
-        response, you must pass its values to the FormatResponse function before
-        responding to the user. 
+        The following function description was also provided:
         
-        {% if basemodel_response -%}
-        `FormatResponse` has the same signature as the function.
-        {% else -%}
-        `FormatResponse` requires keyword arguments, so pass your response under
-        the `data` parameter for validation.
-        {% endif %}
-        """),
-    prompt_library.User(content="""
-        {% if input_binds %} 
+        {{ description }}
+        {% endif %}\
+    """
+    description: str = ""
```

Comment:
Are these attributes extraneous? It looks like they are templated in from the actual AIFn class. I *think* the `Message`-based prompts automatically attempt to pass their own attributes as render kwargs so may be better to remove them to avoid accidental overwriting

Reply:
Ah I see that AIFunction reimplements e.g. source_code as a property